.chloggen/fix-filelog-max-log-size-truncate.yaml

```diff
@@ -0,0 +1,30 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. receiver/filelog)
+component: receiver/filelog
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Fix `max_log_size` behavior to truncate oversized log entries and drop the remainder instead of splitting them into multiple log entries"
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [44371]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  The `max_log_size` setting now correctly truncates log entries that exceed the maximum size and drops the remainder, aligning the implementation with the documented behavior.
+  Previously, oversized log entries were incorrectly split into multiple log entries, which could lead to unexpected log processing behavior.
```

Comment:
I understand this is the desired behavior but, since before was different and not "breaking" anything... should this be considered a breaking change? I'm not 100% sure.